### ISSUE TITLE
DiscoveryNodeRoleIT.testExplicitlyHasAdditionalRole fails on 7.x

### ISSUE THREAD

**Issue by pgomulka** [elastic/elasticsearch#43223]
The test is failing on 7.x with following stacktrace:
```
java.lang.AssertionError:
--
Expected: a collection containing <DiscoveryNodeRole{roleName='additional', roleNameAbbreviation='a'}>
but: mismatches were: [was <DiscoveryNodeRole{roleName='ingest', roleNameAbbreviation='i'}>, was <DiscoveryNodeRole{roleName='data', roleNameAbbreviation='d'}>, was <DiscoveryNodeRole{roleName='additional', roleNameAbbreviation='a'}>, was <DiscoveryNodeRole{roleName='master', roleNameAbbreviation='m'}>]
at __randomizedtesting.SeedInfo.seed([DE1C54CA25478013:CEC687B84041F96F]:0)
at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
at org.junit.Assert.assertThat(Assert.java:956)
at org.junit.Assert.assertThat(Assert.java:923)
at org.elasticsearch.cluster.node.DiscoveryNodeRoleIT.runTestNodeHasAdditionalRole(DiscoveryNodeRoleIT.java:97)
at org.elasticsearch.cluster.node.DiscoveryNodeRoleIT.testExplicitlyHasAdditionalRole(DiscoveryNodeRoleIT.java:80)
```

https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:'2019-06-11T22:00:00.000Z',mode:absolute,to:'2019-06-15T21:59:59.999Z'))&_a=(columns:!(_source),index:b646ed00-7efc-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:DiscoveryNodeRoleIT),sort:!(process.time-start,desc))

one of the failed build:
https://scans.gradle.com/s/erubw2qjmdbak/console-log#L1284

Reproduce line - reproduces locally
```
./gradlew :server:integTest --tests "org.elasticsearch.cluster.node.DiscoveryNodeRoleIT.testExplicitlyHasAdditionalRole" -Dtests.seed=DE1C54CA25478013 -Dtests.security.manager=true -Dtests.locale=ar-IQ -Dtests.timezone=Australia/NSW -Dcompiler.java=12 -Druntime.java=8
```

**Comment by pgomulka** [comment]
is passes for this seed
```
./gradlew :server:integTest --tests "org.elasticsearch.cluster.node.DiscoveryNodeRoleIT.testExplicitlyHasAdditionalRole" -Dtests.seed=5DBF6BADEB4F9F41 -Dtests.security.manager=true -Dtests.locale=ar-IQ -Dtests.timezone=Australia/NSW -Dcompiler.java=12 -Druntime.java=8
```

**Comment by pgomulka** [comment]
@jasontedor would you be able to have a look into this test?

**Comment by pgomulka** [comment]
closed by #43225
